pkg/webhook/resources/virtualmachine/mutator.go

```diff
@@ -310,7 +310,9 @@ func generateMemoryPatch(vm *kubevirtv1.VirtualMachine, mem *resource.Quantity,
 
 	if vm.Spec.Template.Spec.Domain.Memory == nil {
 		patchOps = append(patchOps, fmt.Sprintf(`{"op": "replace", "path": "/spec/template/spec/domain/memory", "value": {"guest":"%s"}}`, &guestMemory))
-	} else {
+	} else if useReservedMemory {
```

Comment:
Technically, we only need to reduce the guest memory if `guest_memory + reserved_memory > limits_memory`. E.g., if the pod's memory limit is to 4GiB, and the guest asks for 3GiB and reserved memory is set to 256MiB. There is no need to reduce the guest value. With this PR, the validator would have rejected such request. I wonder if we actually still need this conditional.